src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -0,0 +1,547 @@
+const debug = Debug('prisma:client:napi')
+
+const MAX_REQUEST_RETRIES = process.env.PRISMA_CLIENT_NO_RETRY ? 1 : 2
+type QueryEngineLogLevel = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'off'
+type QueryEngineEvent = QueryEngineLogEvent | QueryEngineQueryEvent
+type QueryEngineConfig = {
+  datamodel: string
+  datasourceOverrides?: Record<string, string>
+  logLevel: QueryEngineLogLevel
+}
+type QueryEngineLogEvent = {
+  level: string
+  module_path: string
+  message: string
+}
+type QueryEngineQueryEvent = {
+  level: 'info'
+  module_path: string
+  query: string
+  item_type: 'query'
+  params: string
+  duration_ms: string
+  result: string
+}
+
+export interface QueryEngineConstructor {
+  new (
+    config: QueryEngineConfig,
+    logger: (err: string, log: string) => void,
+  ): QueryEngine
+}
+
+type ConnectArgs = {
+  enableRawQueries: boolean
+}
+
+export type QueryEngine = {
+  connect(connectArgs: ConnectArgs): Promise<void>
+  disconnect(): Promise<void>
+  getConfig(): Promise<GetConfigResult>
+  dmmf(): Promise<DMMF.Document>
+  query(request: any): Promise<string>
+  sdlSchema(): Promise<string>
+  serverInfo(): Promise<string>
+  nextLogEvent(): Promise<string>
+}
+
+type ServerInfo = {
+  commit: string
+  version: string
+  primaryConnector: string
+}
+
+type SyncRustError = {
+  is_panic: boolean
+  message: string
+  meta: {
+    full_error: string
+  }
+  error_code: string
+}
+
+type RustRequestError = {
+  is_panic: boolean
+  message: string
+  backtrace: string
+}
+function isQueryEvent(event: QueryEngineEvent): event is QueryEngineQueryEvent {
+  return event.level === 'info' && event['item_type'] === 'query'
+}
+
+const knownPlatforms: Platform[] = [
+  'native',
+  'darwin',
+  'debian-openssl-1.0.x',
+  'debian-openssl-1.1.x',
+  'linux-arm-openssl-1.0.x',
+  'linux-arm-openssl-1.1.x',
+  'rhel-openssl-1.0.x',
+  'rhel-openssl-1.1.x',
+  'linux-musl',
+  'linux-nixos',
+  'windows',
+  'freebsd11',
+  'freebsd12',
+  'openbsd',
+  'netbsd',
+  'arm',
+]
+export class NAPIEngine implements Engine {
+  private engine?: QueryEngine
+  private setupPromise?: Promise<void>
+  private connectPromise?: Promise<void>
+  private config: EngineConfig
+  private QueryEngine?: QueryEngineConstructor
+  private logEmitter: EventEmitter
+  libQueryEnginePath?: string
+  platform?: Platform
+  datasourceOverrides: Record<string, string>
+  datamodel: string
+  logQueries: boolean
+  connected: boolean
+  beforeExitListener?: (args?: any) => any
+
+  constructor(config: EngineConfig) {
+    this.datamodel = fs.readFileSync(config.datamodelPath, 'utf-8')
+    this.config = config
+    this.connected = false
+    this.logQueries = config.logQueries ?? false
+    this.logEmitter = new EventEmitter()
+    this.datasourceOverrides = this.config.datasources
+      ? this.convertDatasources(this.config.datasources)
+      : {}
+
+    if (this.logQueries) {
+      process.env.LOG_QUERIES = 'y'
```

Comment:
In NodeEngine value is 'true', any reason to be different or could we have true there too?

Reply:
There is no difference, but I'll change it to `true`